app/javascript/spec/ansibleCatalog/ansibleCatalogItemForm.test.jsx

```diff
@@ -0,0 +1,183 @@
+      loadDialogs: jest.fn(),
+      loadRepos: jest.fn(),
+      loadCatalogItem: jest.fn(),
+      loadPlaybooks: jest.fn(),
+      loadCredentials: jest.fn(),
+      loadCloudTypes: jest.fn(),
+      loadCloudCredential: jest.fn(),
+      duplicateDropdowns: jest.fn(),
+      region: 10,
+    };
+  });
+  afterEach(() => {
+    fetchMock.reset();
+    fetchMock.restore();
+  });
+  it('should render correctly', () => {
+    const wrapper = shallowMount();
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  it('should allow a catalog item to be edited', () => {
+    const catalogItemId = 1234;
+
+    const loadCatalogItem = jest.fn(() => Promise.resolve({}));
+    const loadCredential = jest.fn(() => Promise.resolve({}));
+    mount(
+      <AnsibleCatalogItemForm1
+        {...initialProps}
+        ansibleCatalog={{ ...reduxStore }}
+        loadCatalogItem={loadCatalogItem}
+        loadCloudCredential={loadCredential}
+        catalogItemFormId={catalogItemId}
+      />,
+      {
+        context: { store },
+        childContextTypes: { store: PropTypes.object.isRequired },
+      },
+    );
+    expect(loadCatalogItem).toHaveBeenCalledWith(1234);
+  });
+  it('should set default values when a catalog item is being edited', (done) => {
+    const catalogItemId = 1234;
+
+    store = mockStore({
+      ansibleCatalog: {
+        ...reduxStore,
+      },
+    });
+    const loadCatalogItem = jest.fn(() => Promise.resolve({}));
+    const loadCloudCredential = jest.fn(() => Promise.resolve({}));
+    const loadPlaybooks = jest.fn(() => Promise.resolve({}));
+    const props = {
+      ...initialProps, loadCatalogItem, loadCloudCredential, loadPlaybooks,
+    };
+    const wrapper = mount(
+      <AnsibleCatalogItemForm1
+        {...props}
+        ansibleCatalog={{ ...reduxStore }}
+        catalogItemFormId={catalogItemId}
+      />
+      , {
+        context: { store },
+        childContextTypes: { store: PropTypes.object.isRequired },
+      },
+    );
+
+    expect(loadCatalogItem).toHaveBeenCalledWith(1234);
+    wrapper.update();
+    setImmediate(() => {
+      expect(loadCloudCredential).toHaveBeenCalledWith(123);
+      const formValues = wrapper.state().initialValues;
+      expect(formValues).toEqual(sampleForm);
+      done();
+    });
+  });
+  it('should allow a form to be submitted', () => {
+    const wrapper = shallowMount();
+    window.location.assign = jest.fn();
+    fetchMock.postOnce('/api/service_templates/', {});
+    const values = {
+      name: 'test',
+      description: '',
+      long_description: '',
+      service_template_catalog_id: '',
+      provision_extra_vars: [],
+      retirement_extra_vars: [],
+    };
+    wrapper.instance().submitForm(values);
+    expect(fetchMock.called('/api/service_templates/')).toBe(true);
+  });
+  it('should report an error if submitting failed', (done) => {
+    // eslint-disable-next-line no-console
+    console.error = jest.fn();
```

Comment:
this is done 9 times in the specs

but this probably breaks `console.error` in *all* the specs

can we use `spyOn` instead?

Reply:
I think i know why this was done. It was done to _silence_ the console errors from appearing when we mock 500 response from API. I will just remove it.